hermes/bin/src/runtime/extensions/mod.rs

```diff
@@ -0,0 +1,150 @@
+//! Hermes runtime extensions
+//!
+//! *Note*
+//! Inspect the generated code with:
+//! ```
+//! cargo expand --bin hermes runtime::extensions
+//! ```
+#![allow(clippy::indexing_slicing)]
+
+use wasmtime::{
+    component::{bindgen, Component, Linker},
+    Config, Engine, Store,
+};
+
+use self::{
+    exports::hermes::cardano::event_on_block::BlockSrc,
+    hermes::{
+        cardano::api::{CardanoBlock, CardanoBlockchainId, CardanoTxn},
+        cron::api::CronTagged,
+    },
+};
+use crate::{runtime, wasm::context::Context};
+
+bindgen!({
+    world: "hermes",
+    path: "../../wasm/wasi/wit",
+});
+
+/// All Hermes runtime extensions states need to implement this.
+pub(crate) trait NewState {
```

Comment:
I'd like to propose renaming the `NewState` trait to `Stateful`. 
While `NewState` is technically accurate, `Stateful` provides several benefits:

**Improved Semantics:** The term `Stateful` more accurately captures the essence of what this trait represents — a component endowed with a state. It goes beyond the initialization aspect, encompassing the broader concept of state management.

**Enhanced Readability:** `Stateful` is straightforward and intuitively understood, making our code more accessible to new contributors and easier to grasp at a glance.

**Future Flexibility:** The name `Stateful` allows for the potential expansion of the trait's responsibilities beyond merely creating a `new` `state`, without necessitating a future rename.

Reply:
All good points.  Thankyou.

Reply:
Done